elastic/apm-server
# Add terraform module to benchmark Elasticsearch on Elastic Cloud with locally built APM-Server and Integration package

## Motivation/summary

Allows running rally to benchmark a locally built Elastic Cloud cluster. This allows developers to benchmark changes introduced in ES schema used by APM-Server.

## Checklist

~~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~~
~~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~~
~~- [ ] Documentation has been updated~~

## How to test these changes

1. Navigate to `testing/rally-cloud` and run `make init` if you are running for the first
time
2. Get `EC_API_KEY` by following [this guide](https://www.elastic.co/guide/en/cloud-enterprise/current/ece-restful-api-authentication.html#ece-api-keys).
3. Checkout the `testing/rally-cloud/variables.tf` to customize your benchmark.
4. Run `EC_API_KEY=<ec_api_key> make apply` to create the required infra and run rally.

The rally benchmark will execute everytime `make apply` is called however, the infrastucture
will be created only once. After testing is done make sure to destroy all infrastructure using
`EC_API_KEY=<ec_api_key> make destroy`


## Related issues

Closes #8754
